appc/appconnector.go

```diff
@@ -400,3 +481,16 @@ func (e *AppConnector) addDomainAddrLocked(domain string, addr netip.Addr) {
 func compareAddr(l, r netip.Addr) int {
 	return l.Compare(r)
 }
+
+// routesWithout returns a without b where a and b
+// are unsorted slices of netip.Prefix
+func routesWithout(a, b []netip.Prefix) []netip.Prefix {
+	m := make(map[netip.Prefix]bool)
+	for _, p := range a {
+		m[p] = true
+	}
+	for _, p := range b {
+		delete(m, p)
+	}
+	return xmaps.Keys(m)
+}
```

Comment:
The number of entries is known already, so providing that to `make` is helpful for reducing allocations.
```suggestion
func routesWithout(a, b []netip.Prefix) []netip.Prefix {
	m := make(map[netip.Prefix]bool, len(a))
	for _, p := range a {
		m[p] = true
	}
	for _, p := range b {
		delete(m, p)
	}
	return xmaps.Keys(m)
}
```

Another option that will avoid shuffling the elements in the slice:
```suggestion
func routesWithout(a, b []netip.Prefix) []netip.Prefix {
	m := make(map[netip.Prefix]bool, len(b))
	for _, p := range b {
		m[p] = true
	}
	return slicesx.Filter(make([]netip.Prefix, 0, len(a)), a, func (p netip.Prefix) bool {
		return !m[p]
	})
}
```